File: app/src/main/java/br/geanezini/mobile/kotlin_room/database/AppDatabase.java
package br.geanezini.mobile.kotlin_room.database;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

import br.geanezini.mobile.kotlin_room.dao.PersonDAO;
import br.geanezini.mobile.kotlin_room.model.Person;

@Database(entities = {Person.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase{

    private static AppDatabase INSTANCE;

    public abstract PersonDAO personDAO();

    public static  AppDatabase getAppDatabase(Context context) {
        if (INSTANCE == null) {
            INSTANCE = Room.databaseBuilder(
                    context.getApplicationContext(),
                    AppDatabase.class,
                    "room-database")
                    .build();
        }

        return INSTANCE;
    }

    public static void destroyInstance() {
        INSTANCE = null;
    }

}

File: app/src/main/kotlin/br/geanezini/mobile/kotlin_room/database/AppDatabase.kt
package br.geanezini.mobile.kotlin_room.database

import android.arch.persistence.room.Database
import android.arch.persistence.room.RoomDatabase
import br.geanezini.mobile.kotlin_room.dao.PersonDAO
import br.geanezini.mobile.kotlin_room.model.Person

@Database(entities = arrayOf(Person::class), version = 1)
abstract class AppDatabase : RoomDatabase() {
    abstract fun personDAO(): PersonDAO
}